test/getGrist.ts

```diff
@@ -196,18 +227,26 @@ export class GristUtils extends GristWebDriverUtils {
     await this.waitForServer();
   }
 
-  public async setCustomWidgetAccess(option: "none"|"read table"|"full") {
+  public async setCustomWidgetAccess(option: "none" | "read table" | "full") {
     const text = {
-      "none" : "No document access",
+      "none": "No document access",
       "read table": "Read selected table",
       "full": "Full document access"
     };
     await this.driver.find(`.test-config-widget-access .test-select-open`).click();
     await this.driver.findContent(`.test-select-menu li`, text[option]).click();
   }
 
-  public async setCustomWidgetMapping(name: string, value: string|RegExp) {
-    const click = (selector: string) => driver.findWait(`${selector}`, 2000).click();
+  public async setCustomWidgetMapping(name: string, value: string | RegExp) {
+    const click = (selector: string) => {
+      try {
+        driver.findWait(`${selector}`, 2000).click();
```

Comment:
Remember this this expression (and the copy below) evaluates to a promise, it is important for it to be awaited somewhere.

Also (I know this is just copied code) `` `${selector}` `` can be written as just `selector`. 

I looked into this and during the period when Grist is setting up the custom widget mapping, the UI can be in flux, and it is not obvious what condition to wait for in order for it to be stable. I wonder if @berhalak might have some advice. The code in this area is based on the `test/nbrowser/CustomWidgetsConfig.ts` test in `grist-core`, but those tests are for specific widgets - there is a `waitForFrame` method there that checks inside the widgets for a "ready" marker. @berhalak do you know if there's a way on the Grist side to know that the widget has called `ready()` and all mapping configuration has been set up?

Reply:
I don't think there is an easy way to detect it. The proper way, I think, would be to modify Grist, and add a testId to the iframe once the ready event is handled (somewhere in WidgetFrame.ts).

Reply:
based on @berhalak message i suggest to put adding testid to the iframe as a separate item in backlog.

Reply:
added